driftbase/api/lobbies.py

```diff
@@ -0,0 +1,281 @@
+"""
+Custom game lobbies for private/direct matches
+"""
+
+from flask_smorest import Blueprint
+from drift.core.extensions.urlregistry import Endpoints
+from marshmallow import Schema, fields
+from flask.views import MethodView
+from flask import url_for, request, abort, make_response, jsonify
+from drift.core.extensions.jwt import current_user
+from driftbase import lobbies
+import http.client as http_client
+import logging
+
+bp = Blueprint("lobbies", "lobbies", url_prefix="/lobbies", description="Custom game lobbies for private/direct matches.")
+endpoints = Endpoints()
+log = logging.getLogger(__name__)
+
+def drift_init_extension(app, api, **kwargs):
+    api.register_blueprint(bp)
+    endpoints.init_app(app)
+
+class CreateLobbyRequestSchema(Schema):
+    team_capacity = fields.Integer(required=True, metadata=dict(description="How many members can be in one team."))
+    team_names = fields.List(fields.String(), required=True, metadata=dict(description="The unique names of the teams."))
+    lobby_name = fields.String(required=False, metadata=dict(description="Optional initial name of the lobby."))
+    map_name = fields.String(required=False, metadata=dict(description="Optional initial map name for the lobby."))
+    custom_data = fields.String(required=False, metadata=dict(description="Optional custom data for the lobby. Will be forwarded to the match server"))
```

Comment:
The custom data is intended to be just that. A field the lobby system cares nothing about.

We could add some size restriction later on if some nefarious no-good-doers decide to abuse this.
Validation however is way trickier since this could be JSON, YAML, Plain text, XML, HTML, Markdown, base64, etc...

Reply:
Will add some sort of limiter